src/Time/TimeSequence.hpp

```diff
@@ -0,0 +1,138 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <optional>
+#include <pup.h>
+#include <vector>
+
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \ingroup TimeSteppersGroup
+///
+/// Holds all the TimeSequences
+namespace TimeSequences {
+template <typename T>
+class EvenlySpaced;
+template <typename T>
+class Specified;
+}  // namespace TimeSequences
+
+/// Represents a sequence of times.
```

Comment:
Could you explain how the template parameter `T` is used in the documentation? E.g. do I choose it to be a `double`, `size_t`, `int`, `Time`, etc? And in what cases would I want to choose which type